client/my-sites/site-settings/site-setting-privacy/form.tsx

```diff
@@ -178,16 +194,14 @@ const SiteSettingPrivacyForm = ( {
 							<FormInputCheckbox
 								name="blog_public"
 								value="0"
-								checked={
-									( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
-									( 0 === blogPublic && ! wpcomPublicComingSoon )
-								}
+								checked={ discourageSearchChecked }
 								onChange={ () =>
 									handleVisibilityOptionChange( {
```

Comment:
Does this function actually handle the `wpcom_data_sharing_opt_out` value? It doesn't seem so.